tensorflow_io/core/plugins/hdfs/hadoop_filesystem.cc

```diff
@@ -0,0 +1,919 @@
+                                       hdfs_file->hdfs_path.c_str());
+      }
+    } else {
+      cur_pos += w;
+    }
+  }
+  TF_SetStatus(status, TF_OK, "");
+}
+
+int64_t Tell(const TF_WritableFile* file, TF_Status* status) {
+  auto hdfs_file = static_cast<HDFSWritableFile*>(file->plugin_file);
+  int64_t position =
+      hdfs_file->libhdfs->hdfsTell(hdfs_file->fs, hdfs_file->handle);
+  if (position == -1)
+    TF_SetStatusFromIOError(status, errno, hdfs_file->hdfs_path.c_str());
+  else
+    TF_SetStatus(status, TF_OK, "");
+  return position;
+}
+
+void Flush(const TF_WritableFile* file, TF_Status* status) {
+  auto hdfs_file = static_cast<HDFSWritableFile*>(file->plugin_file);
+  if (hdfs_file->libhdfs->hdfsHFlush(hdfs_file->fs, hdfs_file->handle) != 0)
+    TF_SetStatusFromIOError(status, errno, hdfs_file->hdfs_path.c_str());
+  else
+    TF_SetStatus(status, TF_OK, "");
+}
+
+void Sync(const TF_WritableFile* file, TF_Status* status) {
+  auto hdfs_file = static_cast<HDFSWritableFile*>(file->plugin_file);
+  if (hdfs_file->libhdfs->hdfsHSync(hdfs_file->fs, hdfs_file->handle) != 0)
+    TF_SetStatusFromIOError(status, errno, hdfs_file->hdfs_path.c_str());
+  else
+    TF_SetStatus(status, TF_OK, "");
+}
+
+void Close(const TF_WritableFile* file, TF_Status* status) {
+  auto hdfs_file = static_cast<HDFSWritableFile*>(file->plugin_file);
+  TF_SetStatus(status, TF_OK, "");
+  if (hdfs_file->libhdfs->hdfsCloseFile(hdfs_file->fs, hdfs_file->handle) != 0)
+    TF_SetStatusFromIOError(status, errno, hdfs_file->hdfs_path.c_str());
+  hdfs_file->fs = nullptr;
+  hdfs_file->handle = nullptr;
+}
+
+}  // namespace tf_writable_file
+
+// SECTION 3. Implementation for `TF_ReadOnlyMemoryRegion`
+// ----------------------------------------------------------------------------
+namespace tf_read_only_memory_region {
+void Cleanup(TF_ReadOnlyMemoryRegion* region) {}
+
+const void* Data(const TF_ReadOnlyMemoryRegion* region) { return nullptr; }
+
+uint64_t Length(const TF_ReadOnlyMemoryRegion* region) { return 0; }
+
+}  // namespace tf_read_only_memory_region
+
+// SECTION 4. Implementation for `TF_Filesystem`, the actual filesystem
+// ----------------------------------------------------------------------------
+namespace tf_hdfs_filesystem {
+
+typedef struct HadoopFileSystemImplementation {
+  LibHDFS* libhdfs;
+  absl::Mutex connection_cache_lock;
+  std::map<std::string, hdfsFS> connection_cache
+      ABSL_GUARDED_BY(connection_cache_lock);
+  HadoopFileSystemImplementation(TF_Status* status);
+  ~HadoopFileSystemImplementation() {
+    if (libhdfs != nullptr) {
+      delete libhdfs;
+    }
+  }
+} HadoopFileSystemImplementation;
+
+HadoopFileSystemImplementation::HadoopFileSystemImplementation(
+    TF_Status* status)
+    : libhdfs(new LibHDFS(status)),
+      connection_cache_lock(),
+      connection_cache() {}
+
+typedef struct HadoopFileSystem {
+  absl::Mutex mu;
+  std::unique_ptr<HadoopFileSystemImplementation> ptr ABSL_GUARDED_BY(mu);
```

Comment:
I think the test failed because you forgot to add `#include<memory>`

Reply:
@vnvo2409 Thanks. The PR has been updated.